### PR TITLE
Fix kubectl sort-by metadata issue.

### PR DESCRIPTION
**Pull request by aveshagarwal**

### CODE REVIEW

File: pkg/kubectl/resource_printer.go [hunk 1]
```diff
@@ -424,6 +425,8 @@ var podSecurityPolicyColumns = []string{"NAME", "PRIV", "CAPS", "VOLUMEPLUGINS",
 func (h *HumanReadablePrinter) addDefaultHandlers() {
 	h.Handler(podColumns, printPod)
 	h.Handler(podColumns, printPodList)
+	h.Handler(podColumns, printPodv1)
```
Comment (pmorie) [inline pkg/kubectl/resource_printer.go]: This feels wrong to me

Comment (ncdc) [inline pkg/kubectl/resource_printer.go]: Ditto. We ideally shouldn't need version specific printers.

**Comment by aveshagarwal** [comment]
@pmorie thanks. then how to print versioned objects? convert versioned objects again to internal types and print? But I am not sure how to do that in the code? any pointers?

**Comment by pmorie** [comment]
@deads2k PTAL

**Comment by ncdc** [comment]
Why do you need the json metadata to do the sorting?

**Comment by pmorie** [comment]
I think that we need to apply sorting on versioned schema, but we should print the same stuff we print today.  To be fair, @aveshagarwal, I think this is a tricky issue to sort out :)

**Comment by aveshagarwal** [comment]
As per https://github.com/kubernetes/kubernetes/blob/master/docs/user-guide/kubectl-overview.md:

```
kubectl [command] [TYPE] [NAME] --sort-by=<jsonpath_exp>
```

And so json metadata used by sort-by.

**Comment by ncdc** [comment]
Sure, but is there code that looks at the json tags to do the sorting?

**Comment by aveshagarwal** [comment]
yes in pkg/kubectl/sorting_printer.go.

**Comment by aveshagarwal** [comment]
I am trying another approach where instead of printing a versioned object, I will try to print its corresponding internal object as sorter keeps original position of objects too.

**Comment by aveshagarwal** [comment]
That seem to work (feeling why I did not think it before), will do more testing with origin and kube before submitting to kube upstream

**Comment by aveshagarwal** [comment]
Updated PR, with some testing will send to upstream unless there is some more feedback. Probably thought too much about versioned objects that obscured this simple solution.

File: pkg/kubectl/cmd/get.go [hunk 2]
```diff
@@ -262,10 +278,13 @@ func RunGet(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string
 
 	for ix := range objs {
 		var mapping *meta.RESTMapping
+		var tmp runtime.Object
```
Comment (ncdc) [inline pkg/kubectl/cmd/get.go]: Call this something more informative that `tmp` that describes its purpose

**Comment by aveshagarwal** [comment]
I am closing this as I have sent it to kube upstream.

**Pull request closed without merging**